### ISSUE TITLE
base64: invalid input

### ISSUE THREAD

**Issue by Daenara** [Boteium/bashbullet#7]
I just installed bashbullet and tried to get it working. I put the API key in the config and started in on a terminal. Whenever I receive a message vie pushbullet I just get "base64: invalid input". I tried sending myself messages, those don't even show  on the terminal. I haven't seen any popup appear even thought notify-send works from terminal. Opening the graphical interface I get "Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged." but it opens and I can send pushmessages out. I'm not sure if I did something wrong with the setup (installed via AUR, no end to end encryption) or if it's just deprecated or not working anymore. Any help would be appreciated since I really want to receive pushbullet messages on desktop without using chrome.

**Comment by Boteium** [comment]
Do you have a notification daemon installed and running on your system ?  (For example : notify-osd / xfce4-notifyd / dunst / lxqt-notificationd )

**Comment by Daenara** [comment]
Yes, I do have one. I had it working but it seems it crashed. After repairing that I do get notification popups but the message "base64: invalid input" stays.

**Comment by Boteium** [comment]
So popup of push works now. How about notification mirroring ? Does it work ? Does application icon shows alongside of notification text. base64 is merely for decoding app icon only (when mirroring notifications rather than pushes) . You can send a test notification from pushbullet's "notification mirroring" setup menu on your phone.

**Comment by Daenara** [comment]
Mirroring works fine, the icon does show up but the message stays. As long as it works I can live with the message but there must be a reason for it to show up.

**Comment by Boteium** [comment]
Probably because I didn't handle the newline character properly. But base64 does work correctly, otherwise you won't see any app icon. I'm not interested in fixing this glitch right now.

**Comment by Boteium** [comment]
Is it a push or a mirroring notification ? Only pushes will end up in inbox.

**Comment by Daenara** [comment]
I'm not really sure what the difference between the notifications is but I don't get any notifications in the inbox. I'm using bashbullet to show notifications on my desktop pc so I'm pretty sure that I get pushnotifications sometimes. When I start bashbullet the inbox shows a message telling me that there are no new messages but after letting it run for a while and receiving notifications that string disapears and I'm quite sure thet there should be notifications shown in there. I let it run from terminal for a while but I got no errormessages other than the base64 one and the gtk warning.

**Comment by Boteium** [comment]
A push is a link/text/file you send using Pushbullet's website or Pushbullet's smartphone application. A mirroring notification is a notification appears on your phone, which Pushbullet will duplicate and send to other device ( like bashbullet ). Notifications are "ephemeral". Pushes are not and would be saved in both pushbullet's cloud and bashbullet's inbox.

**Comment by Daenara** [comment]
Now that I'm home I could test the inbox a bit more. First I sent a pushmessage from my phone to the already running bashbullet and it did exactly nothing, no notification popping up and no change in the inbox icon. Since I suspend my PC it could be an issue with screen and suspend to ram so I killed all bashbullet processes and started a new one. There I immediately saw the icon for new messages but 
there was nothing there. Here is a screenshot of that area: 
![bashbullet_0](https://cloud.githubusercontent.com/assets/3619315/15516136/bd95aa52-21f1-11e6-96af-580dcca9d0db.png)

I then clicked mark all as read an the icon returned to normal:
![bashbullet_1](https://cloud.githubusercontent.com/assets/3619315/15516152/cfded076-21f1-11e6-9479-b38ba5211459.png)

All the terminal said was the Gtk warning, no error message or something.

**Comment by Boteium** [comment]
Can you check if there are saved message in bashbullet profile folder for me ?
ls $HOME/.bashbullet/received/

**Comment by Daenara** [comment]
The received folder is empty, only the archive folder exists there. In there are 4 files all named like "1464115543.5422709_bashbullet_dismissed" but those are empty. Also there is an OldArchive.txt which contains "No archived messages"

**Comment by Boteium** [comment]
This is expected behavior. Dismissed message (Maybe a notification or deleted message) would not be displayed and *_bashbullet_dismissed would be deleted later. It's also a known issue that wscat would have problem over unstable connection or suspend/hibernation. I haven't come with a mechanism to restart wscat automatically. But what I can't figure out is why the push you sent didn't show up in your inbox.

Can you send a push again while bashbullet is running.

**Comment by Daenara** [comment]
So, I did send a few pushes again and watched what happened in the received folder. There were a few files still there but after every push I sent another file appeared. Here the folders content:

1. 1464784604.5823629_all_dismissed  
    1464961171.147723_all_dismissed  
    1465062769.1428869_bashbullet_dismissed  
    archived

2. 1464784604.5823629_all_dismissed  
    1464961171.147723_all_dismissed  
    1465062769.1428869_bashbullet_dismissed  
    1465062879.5874031_bashbullet_dismissed  
    archived

3. 1464784604.5823629_all_dismissed  
    1464961171.147723_all_dismissed  
    1465062769.1428869_bashbullet_dismissed  
    1465062879.5874031_bashbullet_dismissed  
    1465062905.3005829_bashbullet_dismissed  
    archived

It seems new pushes just get dismissed by default right now

**Comment by Boteium** [comment]
OK, Thanks for you information. Do you have any skip_X such as skip_0="true" or skip_1="true" (and so on) in~/.bashbullet/config ? If you don't have them I guess there might be problem when parsing config.

**Comment by Daenara** [comment]
I don't have those, the only thing in the config is the api key

**Comment by Boteium** [comment]
can you comment out line 285 and 287 288 289 in /usr/bin/bashbullet and try again ? Just to make sure if my guess is correct.
[bashbullet](https://github.com/Boteium/bashbullet/blob/master/bashbullet)

**Comment by Daenara** [comment]
Okay, I did that. Running bashbullet without -scr and sending a push with a link gave me 1465065320.130924_bashbullet_link which contains the push, but it is still not showing up in the window. Running with -scr and with a text only push gave me 1465065489.67308_bashbullet_note but I still can't open the window when started with -scr

**Comment by Daenara** [comment]
Okay, it seems that the window thing was a not properly closed bashbullet process, when opening it without -scr it also shows all messages which already have files, but it doesn't updates the messages

**Comment by Boteium** [comment]
Can you be more specific about the "update the message" part ? BTW, the inbox windows only aggregate all the push upon left click on systray icon. So, It will new message while inbox is still open will not be shown immediately.

I also noticed that the screenshot of inbox you provided two weeks ago is empty. But if there is no new messages it should have a "No new messages" message in it. So, there might be more bugs.

**Comment by Daenara** [comment]
Okay, if the inbox isn't supposed to show new messages immediately then there is no update issue.

The inbox did display the "No new message" string until I did send a push message, then the window was empty. My guess is that it should have displayed the new push there but it failed and the string was overwritten with an empty one

**Comment by Boteium** [comment]
Thanks for your help ! It's a error I made. It's because the script would count dismissed message as "new message". As for the "dismissed by default" issue, I still have no clue why. I will look into it.

**Comment by Daenara** [comment]
Thanks. I still have one question left, you wrote that bashbullet has to be started using screen/systemd/upstart. I get how it works with screen as I'm using that but how do I get it to start with systemd. I tried and got it started but it only complained that it's running in the background and closed itself. I gave up after a while and installed screen but I'm curious how it works.
